angio-frontend/src/modules/analyse/components/AnalyseListFilter.vue

```diff
@@ -0,0 +1,155 @@
+
+                        <v-flex xs5 pl-2>
+                            <v-checkbox
+                                    v-model="isPeriod"
+                                    v-bind:label="$t('analyse.component.analyseListFilter.period')"
+                            ></v-checkbox>
+                        </v-flex>
+
+                        <v-flex xs2 pl-2>
+                            <v-checkbox
+                                    v-model="analyseFilterModel.isStarred"
+                                    v-bind:label="$t('analyse.component.analyseListFilter.starred')"
+                                    class="justify-end"
+                            ></v-checkbox>
+                        </v-flex>
+                    </v-layout>
+                </v-flex>
+            </v-layout>
+        </v-card-title>
+    </v-card>
+</template>
+
+<script lang="ts">
+    import {Component, Emit, Prop, Vue} from 'vue-property-decorator';
+    import {AnalyseFilterModel, AnalyseStatusType} from '@/modules/analyse/models/analyse';
+    import {Locale} from '@/modules/user/store/userState';
+    import {debounce} from 'helpful-decorators';
+    import PeriodDatePicker from '@/modules/common/components/PeriodDatePicker.vue';
+    import {PeriodDateModel} from '@/modules/common/models/PeriodDateModel';
+    import {AnalyseStatusRepresentation} from '@/modules/analyse/helpers/representation';
+    import {CommonEvent} from '@/modules/common/helpers/commonEvent';
+
+    @Component({
+        components: {PeriodDatePicker},
+    })
+    export default class AnalyseListFilter extends Vue {
+
+        @Prop()
+        public readonly locale!: Locale;
+
+        public statuses: AnalyseStatusRepresentation[] = [
+            { value: AnalyseStatusType.CREATED, text: 'analyse.component.analyseListFilter.status.CREATED' },
+            { value: AnalyseStatusType.SUCCESS, text: 'analyse.component.analyseListFilter.status.SUCCESS' },
+            { value: AnalyseStatusType.IN_PROGRESS, text: 'analyse.component.analyseListFilter.status.IN_PROGRESS' },
+            { value: AnalyseStatusType.FAILED, text: 'analyse.component.analyseListFilter.status.FAILED' }
+        ];
+
+        public selectedStatuses: AnalyseStatusRepresentation[] = [];
+
+        public isPeriod: boolean = true;
+
+        public periodDateModel: PeriodDateModel = {
+            startDate: undefined,
+            endDate: undefined,
+            date: undefined
+        };
+
+        public analyseFilterModel: AnalyseFilterModel = {
+            search: undefined,
+            statuses: [],
+            startDate: undefined,
+            endDate: undefined,
+            date: undefined,
+            isStarred: undefined
+        };
+
+        public onPeriodDataChanged(periodDateModel: PeriodDateModel) {
+            this.periodDateModel = periodDateModel;
+            this.updateDates()
+        }
+
+        public updateDates() {
+            if (this.isPeriod) {
```

Comment:
1. `PeriodDatePicker` must encapsulate logic of output data
2. You can get rid of the extra variable `periodDateModel` by moving mapping to `onPeriodDataChanged`

Reply:
Done